### DISCUSSION TITLE
Project Maintenance

### DISCUSSION THREAD

**Discussion by nlochschmidt** [konform-kt/konform#38] [General]
_@cies reached out via Twitter to ask about the state of this project so I decided to respond here and have it visible._

To be honest, when I started this project back in 2018, I really didn't expect this to see any traction. It was quite a nice surprise seeing that the number of stargazers is approaching the 500 mark 🤗 . 

Remember, this library is very simple. The initial version took about a week to write and as it is it seems to be useful to a nice number of people. Even though the internal design is a little bit rough in some places, I am overall very happy with how extensible it is. This means that in a lot of cases this project doesn't need to accept pull requests for additional built-in validations as people can just add them depending on the needs of their project.

That said, I haven't had a use for the project myself and sadly this didn't change since my last message responding to #12. 

To be clear, I haven't abandoned the project. I am reading all the issues when they come in. My lack of reactions to issues has a couple of reasons/excuses:

   1. The company I work for went into hypergrowth so I have been quite busy (but who isn't these days)
   2. On investigation of the issues, quite a few of them turned out to be difficult to implement. But writing down why they are difficult to implement turned out to also not be easy. Since I wasn't able to respond adequately I often didn't respond at all (mainly because of 1.)
   3. There are issues, but there are not a lot of pull requests for them (other people experiencing 1. and/or 2.?)
 
 That said, I really want to improve the situation. So here is what I am going to do:

- [ ] Write a `CONTRIBUTING.md` that outlines how to best contribute to this project
- Reconsider #32
- Prioritize reviewing open pull requests
- Go through the open issues and check what is still relevant

In the meantime, If you like the project and want to help, my wish would be that you would

- Engage in discussions around issues and propose options to solve them (e.g. thinking about how the DSL/API ideally looks like)
- Prototype some of the proposed options
- Actually open pull requests for the best solution

Also feel free to comment on #32, if you are interested in becoming a co-maintainer for Konform.

Please be kind. The last two years have not been 🙏.  

And thank you for your interest in Konform.

**Comment by nlochschmidt** [comment]
It has been a long time, since the message above, but there is finally good news to share.

For a couple of years now I had hoped to eventually get the chance to use my own library in a work setting, but it never happened. I have personally moved even further away from the Kotlin ecosystem and I can now finally acknowledge, that I won't be getting back to this project. Rather than archiving it and letting it slowly fade away, I've decided to let go. 

This is where @dhoepelman comes in. David contacted me this week and we had a good chat today. **We agreed that he will take over the maintenance and future direction of this library 🎉** . David is actively developing in Kotlin and is using konform in his day job, which I believe will lead to good impulses for this project.

I will stay on as co-maintainer for up to six month, helping with the transition and technical issues that may come up, however effective today, the future direction of this project is in David's hands.

To everyone who is using konform: Thank you! I hope this change will increase your confidence in continuing to use it. I also hope that more activity in the project will lead to more people enjoying what I believe is a pretty nice way of writing validations in Kotlin 😉

**Reply by jillesvangurp** [reply]
Thanks, for your work on this. We are also still using konform. It's great as is but I'd be happy to help out modernizing dependencies, or doing other things as needed.

**Reply by dhoepelman** [reply]
Thanks @jillesvangurp! Since you contributed before, I've given you write access so you can easier create PRs etc.

**Reply by jillesvangurp** [reply]
👍  I just took a quick look at the build. It looks fine as it is. Bringing in more platform support might be useful. But probably better to wait for kotlin 2.0 as they've improved the multi platform build config.

**Comment by dhoepelman** [comment]
@nlochschmidt mind adding @jillesvangurp as a member of konform-kt org? I don't have the rights yet. He has made multiple contributions in the recent weeks.

**Comment by jillesvangurp** [comment]
I think I've already been added. Not sure if I need the privileges. But I'm happy to do some light review work, build updating, etc. once in a while.
